WIP - Compliance Backend Storage

rk-fsdev opened the pull request.

Commit:
Add Compliance Backend Storage

Update archway.mwb with adding compliance_v2 table

Add compliance_v2 table

Update WorkspaceRequest.Compliance

Update compliance repository

Remove Compliance model, update workspace request

Fix some errors

Commit:
Add findWorkspaceById in compliance repository

afoerster commented on common/src/main/scala/io/phdata/repositories/WorkspaceRequestRepositoryImpl.scala:
```diff
@@ -187,7 +179,7 @@ class WorkspaceRequestRepositoryImpl(sqlSyntax: SqlSyntax) extends WorkspaceRequ
             ${workspaceRequest.description},
             ${workspaceRequest.behavior},
-            ${workspaceRequest.compliance.id},
+            0,
```
Can you add a comment here that this field is no longer used?

Commit:
Remove group_id in compliance_v2 table

Commit:
Add selectLegacyFragment for legacy workspaces sql
Remaining tasks:
- Check original compliance table for the legacy workspaces, it means workspace compliance_id is not zero
- There are some other methods which is using legacy workspace fragments in workspace request repository.
- HiveDatabaseRegistrationProvisioning class modification of createDBProperties

martin-morek commented on common/src/main/scala/io/phdata/repositories/ComplianceRepositoryImpl.scala:
```diff
@@ -1,47 +1,37 @@
+
+    def insert(questionId: Long, workspaceId: Long, groupId: Long): Update0 =
+      sql"""
+       insert into compliance (workspace_id, question_id, group_id)
```
I believe it should be new table `compliance_v2` instead of old `compliance`, also I don't think we need `group_id`

afoerster commented:
Closing until we can re-prioritize this feature
